### PR TITLE
[FEATURE] Use headers from plugin response in parent response

### PR DESCRIPTION
**Pull request by dlubitz**
Enables possibility of server-side redirects, triggered within
a plugin e.g. after successful frontend login.

### CODE REVIEW

File: TYPO3.Neos/Classes/TYPO3/Neos/TypoScript/PluginImplementation.php [hunk 1]
```diff
@@ -149,6 +149,12 @@ public function evaluate() {
 			$pluginResponse = new Response($parentResponse);
 
 			$this->dispatcher->dispatch($this->buildPluginRequest(), $pluginResponse);
+			
+      /* Put header from sub-request into main-request */
```
Comment (hlubek) [inline TYPO3.Neos/Classes/TYPO3/Neos/TypoScript/PluginImplementation.php]: With the current Flow CGL the comment should look like:

    // Put header from ...
    if (...) {
    }

Inline comments should generally be avoided and could be replaced by extracting the code into a method with a speaking name.

**Comment by bwaidelich** [comment]
Thanks for this!
The change definitely makes sense but I would like a more generic solution: IMO *every* header of the plugin response should be bubbled up to the main request - do you agree?

**Comment by dlubitz** [comment]
Sure, but I guess this would be a bigger change respectively testing. Also it could be helpful to define a "override strategy" of the headers. Not sure if it always makes sence to override the value of the main request with a value of a child one in all cases.

**Comment by bwaidelich** [comment]
> Also it could be helpful to define a "override strategy" of the headers.

Actually I think it should be really simple. I can't think of a scenario where you don't want to forward a response header. and actually we already do that for widget requests IIRC

> Not sure if it always makes sence to override the value of the main request with a value of a child one in all cases.

Not *request*, just the *response*. The plugin has its own response body so that it can be rendered individually. But the headers are currently "lost"

**Comment by dlubitz** [comment]
Yes, sorry. Response. Ok, I can update the PR in next days. Just setting all plugin response headers into the headers of parent response. Could you give me a hint where to find the "widget requests" stuff? Maybe it is reusable.

**Comment by bwaidelich** [comment]
Awesome, thanks.
I think I was wrong regarding the widget response, but maybe it still helps: https://github.com/neos/flow-development-collection/blob/master/TYPO3.Fluid/Classes/TYPO3/Fluid/Core/Widget/AbstractWidgetViewHelper.php#L201-L246

**Comment by aertmann** [comment]
closed and re-opened in the hope of Travis reporting status back.. sometimes works, although not always

**Comment by aertmann** [comment]
like the feature, however the change of behavior needs to be documented.. also did you consider adding a test for it?

also it is actually potentially breaking, but the risk is negligible IMO so no need to worry about it

**Comment by dlubitz** [comment]
@aertmann: I added my first neos unittest ;) Would be great, if you could have a deeper look into it.

**Comment by aertmann** [comment]
@dlubitz: looks good, just missing documentation now..

test pass btw. https://travis-ci.org/neos/neos-development-collection/builds/91988410

**Comment by dlubitz** [comment]
@aertmann Where do you think it needs to be documentend?

**Comment by aertmann** [comment]
@dlubitz: somewhere in http://neos.readthedocs.org/en/latest/ExtendingNeos/CreatingAPlugin.html maybe?

**Comment by bwaidelich** [comment]
:+1: by reading
@dlubitz awesome, thanks for looking into this. IMO it could even be merged as is because it should be the expected behavior. But even better if you mention it in the docs of course. And I agree that `Creating a plugin` would be a good place for it

**Comment by kitsunet** [comment]
(just wanted to restart travis)...

**Comment by kitsunet** [comment]
:+1:

**Comment by kitsunet** [comment]
Travis is behaving a bit odd but I tested it all locally as well...

**Pull request merged**